### PR TITLE
fix: extract bar chart config component

### PR DESCRIPTION
**Pull request by owlas**
Closes: #10749 

### Description:

Moves the bar chart config to its own slice

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/features/sqlRunner/components/visualizations/BarChartConfiguration.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+const DEBOUNCE_TIME = 500;
+
+const BarChartConfiguration: FC<BarChartConfigurationProps> = ({
+    value,
+    onChange,
+}) => {
+    const onXAxisLabelChange = debounce((label: string) => {
+        onChange({
+            ...value,
```
Comment (magnew) [inline packages/frontend/src/features/sqlRunner/components/visualizations/BarChartConfiguration.tsx:20]: Doing all of this spreading before sending the values to the store seems to me like skirting some of the point of using RTK. You were concerned about having too many specific setters for individual properties, no? But it seems like this way we have moved that complexity out of the slice and now we have to do this error-prone spreads in these configuration files for every property. I don't think this is less code than the way we were doing it before where there were actions per-property, right? And I think it's probably slower and more error prone than passing one property to the store and letting relying on RTK/Immer to set the nested properties. One of the things RTK is good at/good for is surgically alerting object without having to spread/copy the whole thing.

Maybe I missed a reason to do this 😅? If we do want to only have one setter in the store that takes the whole config, maybe we should use Immer here?

File: packages/frontend/src/features/sqlRunner/components/visualizations/BarChart.tsx [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 import { useBarChartDataTransformer } from '../../transformers/useBarChartDataTransformer';
 
 type BarChartProps = {
     data: NonNullable<ReturnType<typeof useSqlQueryRun>['data']>;
-    config: BarChartConfig | undefined;
 } & Partial<Pick<EChartsReactProps, 'style'>>;
 
-const BarChart: FC<BarChartProps> = ({ data, config, style }) => {
+const BarChart: FC<BarChartProps> = ({ data, style }) => {
+    const config = useAppSelector((state) => state.barChartConfig.config);
```
Comment (joaoviana) [inline packages/frontend/src/features/sqlRunner/components/visualizations/BarChart.tsx]: Could you move to the parent component and keep the prop `config` pls? 
So it doesn't break sql charts in dashboards (we don't have RTK in any other part of the app)

File: packages/frontend/src/features/sqlRunner/store/index.ts [hunk 3]
```diff
@@ -1,13 +1,14 @@
 // FIXES ts2742 issue with configureStore
 // eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars
-import type * as rtk from '@reduxjs/toolkit';
```
Comment (joaoviana) [inline packages/frontend/src/features/sqlRunner/store/index.ts:3]: we need to add this back so the build passes pls 👍

File: packages/frontend/src/features/sqlRunner/store/barChartSlice.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                    },
+                    series: fieldIds.slice(1).map((reference, index) => ({
+                        reference,
+                        name: reference,
+                        yIndex: index,
+                    })),
+                };
+            }
+        });
+        builder.addCase(setSaveChartData, (state, action) => {
```
Comment (joaoviana) [inline packages/frontend/src/features/sqlRunner/store/barChartSlice.ts:68]: nice 👍

File: packages/frontend/src/features/sqlRunner/store/barChartSlice.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
+import { ChartKind, type BarChartConfig } from '@lightdash/common';
+import type { PayloadAction } from '@reduxjs/toolkit';
+import { createSlice } from '@reduxjs/toolkit';
+import { setInitialResultsAndSeries, setSaveChartData } from './sqlRunnerSlice';
+
+const initialState: { config: BarChartConfig | undefined } = {
+    config: undefined,
+};
+
+export const barChartSlice = createSlice({
```
Comment (joaoviana) [inline packages/frontend/src/features/sqlRunner/store/barChartSlice.ts:10]:
```suggestion
export const barChartConfigSlice = createSlice({
```

Comment (joaoviana) [reply packages/frontend/src/features/sqlRunner/store/barChartSlice.ts:10]: And if you could rename the file to `barChartConfigSlice` so everything matches 👍

**Review by joaoviana — APPROVED** [review]
Some small requested changes 👍

File: packages/frontend/src/features/sqlRunner/components/visualizations/BarChart.tsx (hunk 2 above)
Comment (joaoviana) [reply packages/frontend/src/features/sqlRunner/components/visualizations/BarChart.tsx]: I believe this is done

File: packages/frontend/src/features/sqlRunner/store/index.ts (hunk 3 above)
Comment (magnew) [reply packages/frontend/src/features/sqlRunner/store/index.ts:3]: WFT, merges today

**Pull request merged**